src/server/auth/server/oidc.go

```diff
@@ -384,42 +340,53 @@ func (o *InternalOIDCProvider) validateIDToken(ctx context.Context, rawIDToken s
 		return nil, nil, errors.Wrapf(err, "could not get claims")
 	}
 
-	if !claims.EmailVerified && !o.IgnoreEmailVerified {
+	if !claims.EmailVerified && !config.IgnoreEmailVerified {
 		return nil, nil, errors.Wrapf(err, "email_verified claim was false")
 	}
 	return idToken, &claims, nil
 }
 
-func (o *InternalOIDCProvider) syncGroupMembership(ctx context.Context, claims *IDTokenClaims) error {
+func (a *apiServer) syncGroupMembership(ctx context.Context, claims *IDTokenClaims) error {
 	groups := make([]string, len(claims.Groups))
 	for i, g := range claims.Groups {
-		groups[i] = fmt.Sprintf("group/%s:%s", o.Prefix, g)
+		groups[i] = fmt.Sprintf("group:%s", g)
 	}
 	// Sync group membership based on the groups claim, if any
-	return o.a.setGroupsForUserInternal(ctx, claims.Email, groups)
+	return a.setGroupsForUserInternal(ctx, claims.Email, groups)
 }
 
 // handleOIDCExchangeInternal is a convenience function for converting an
 // authorization code into an access token. The caller (handleOIDCExchange) is
 // responsible for storing any responses from this in etcd and sending an HTTP
 // response to the user's browser.
-func (a *apiServer) handleOIDCExchangeInternal(ctx context.Context, sp *InternalOIDCProvider, authCode, state string) (nonce, email string, retErr error) {
+func (a *apiServer) handleOIDCExchangeInternal(ctx context.Context, authCode, state string) (nonce, email string, retErr error) {
```

Comment:
Maybe out of scope for this PR, but I just noticed that `state` isn't used and can be removed (or the etcd write could be moved into this function, which is kind of how I expected it to work :shrug:)

Reply:
That method is pretty complicated, I'll make a note to try and simplify it in a future PR so it can get the appropriate level of attention.